source/common/router/router.h

```diff
@@ -303,12 +321,21 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     void onBelowWriteBufferLowWatermark() override { enableDataFromDownstream(); }
 
     void disableDataFromDownstream() {
-      ASSERT(parent_.upstream_requests_.size() == 1);
+      // If there is only one upstream request, we can be assured that
```

Comment:
Can you add additional text here and below and how we know this assert is true in the case of hedging? It might be easier to put this assert in a helper function where the comment can leave, but nbd either way and up to you.